src/deadline_test_fixtures/deadline/worker.py

```diff
@@ -539,30 +428,96 @@ def start_windows_worker(self) -> None:
+            *(config.pre_install_commands or []),
+            # fmt: off
+            (
+                "install-deadline-worker "
+                + "-y "
+                + f"--farm-id {config.farm_id} "
+                + f"--fleet-id {config.fleet.id} "
+                + f"--region {config.region} "
+                + f"--user {config.agent_user} "
+                + f"{'--allow-shutdown ' if config.allow_shutdown else ''}"
+                + "--start"
+            ),
+            # fmt: on
+        ]
+
+        if config.service_model_path:
+            cmds.append(
+                f"aws configure add-model --service-model file://{config.service_model_path} --service-name deadline; "
+                f"Copy-Item -Path ~\\.aws\\* -Destination C:\\Users\\Administrator\\.aws\\models -Recurse; "
+                f"Copy-Item -Path ~\\.aws\\* -Destination C:\\Users\\{config.agent_user}\\.aws\\models -Recurse; "
+                f"Copy-Item -Path ~\\.aws\\* -Destination C:\\Users\\{config.job_user}\\.aws\\models -Recurse"
+            )
 
-        if self.configuration.operating_system.name == "AL2023":
-            self.start_linux_worker()
-        else:
-            self.start_windows_worker()
+        if os.environ.get("DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE"):
+            LOG.info(
+                f"Using DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE: {os.environ.get('DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE')}"
+            )
+            cmds.insert(
+                0,
+                f"[System.Environment]::SetEnvironmentVariable('DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE', '{os.environ.get('DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE')}', [System.EnvironmentVariableTarget]::Machine); "
+                "$env:DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE = [System.Environment]::GetEnvironmentVariable('DEADLINE_WORKER_ALLOW_INSTANCE_PROFILE','Machine')",
+            )
 
-        self.worker_id = self.get_worker_id()
+        if os.environ.get("AWS_ENDPOINT_URL_DEADLINE"):
+            LOG.info(
+                f"Using AWS_ENDPOINT_URL_DEADLINE: {os.environ.get('AWS_ENDPOINT_URL_DEADLINE')}"
+            )
+            cmds.insert(
+                0,
+                f"[System.Environment]::SetEnvironmentVariable('AWS_ENDPOINT_URL_DEADLINE', '{os.environ.get('AWS_ENDPOINT_URL_DEADLINE')}', [System.EnvironmentVariableTarget]::Machine); "
+                "$env:AWS_ENDPOINT_URL_DEADLINE = [System.Environment]::GetEnvironmentVariable('AWS_ENDPOINT_URL_DEADLINE','Machine')",
+            )
+
+        return "; ".join(cmds)
+
+    def userdata(self, s3_files) -> str:
+        copy_s3_command = ""
+        if s3_files:
+            copy_s3_command = " ; ".join([f"aws s3 cp {s3_uri} {dst}" for s3_uri, dst in s3_files])
+
+        userdata = f"""<powershell>
+            Invoke-WebRequest -Uri "https://www.python.org/ftp/python/3.11.9/python-3.11.9-amd64.exe" -OutFile "C:\python-3.11.9-amd64.exe"
```

Comment:
Two things:
1. Patching. This needs to be kept up-to-date. How are we doing that?
2. Supply-chain vulnerability. Python publishes both a GPG and Sigstore signature. We should be verifying the download is authentic in some way before continuing. e.g. Recognizing that a WInServer AMI doesn't have GPG or sigstore apps by default, and so we'd have a chicken & egg problem... even a sha256 check is better than nothing.